### PR TITLE
[Fix #1887] Do not generate row for invalid pids

### LINKED ISSUE #1887: [OS X] Having an unsatisfiable constraint on `pid` column in processes causes an output row
This seems like a bug (mostly cosmetic perhaps):

Adding a `WHERE pid = xxx` constraint on `processes` table (xxx pid doesn't exist) causes a row of output:

```
osquery> select name,pid,path,cmdline,state,uid,gid,on_disk,parent from processes
    ...> where pid = 65432;
+------+-------+------+---------+-------+-----+-----+---------+--------+
| name | pid   | path | cmdline | state | uid | gid | on_disk | parent |
+------+-------+------+---------+-------+-----+-----+---------+--------+
|      | 65432 |      |         | 0     | -1  | -1  | -1      | 0      |
+------+-------+------+---------+-------+-----+-----+---------+--------+
osquery> select name,pid,path,cmdline,state,uid,gid,on_disk,parent from processes
    ...> where pid = 1111;
+------+------+------+---------+-------+-----+-----+---------+--------+
| name | pid  | path | cmdline | state | uid | gid | on_disk | parent |
+------+------+------+---------+-------+-----+-----+---------+--------+
|      | 1111 |      |         | 0     | -1  | -1  | -1      | 0      |
+------+------+------+---------+-------+-----+-----+---------+--------+
osquery> select name,pid,path,cmdline,state,uid,gid,on_disk,parent from processes
    ...> where pid = 123456;
+------+--------+------+---------+-------+-----+-----+---------+--------+
| name | pid    | path | cmdline | state | uid | gid | on_disk | parent |
+------+--------+------+---------+-------+-----+-----+---------+--------+
|      | 123456 |      |         | 0     | -1  | -1  | -1      | 0      |
+------+--------+------+---------+-------+-----+-----+---------+--------+
```

Here the pids are non existent. This only happens on OS X.

Feel free to close if this is as expected.

### PR DESCRIPTION
**Pull request by theopolis**

### CODE REVIEW

**Comment by theopolis** [comment]
With respect to #1886, I think this will prevent Zombie process from showing up in OS X. I am not sure if they are reported in the default process list either way.

**Comment by osqueryer** [comment]
Build origin/pr/1888/merge (Job: 2422) is good! :+1:

**Comment by sharvilshah** [comment]
So fast!

https://i.giphy.com/94SrPuxGICnFC.gif

**Comment by sharvilshah** [comment]
Re: Zombie processes does show up in Apple's default `ps`

File: osquery/tables/system/darwin/processes.cpp [hunk 1]
```diff
@@ -267,16 +267,7 @@ QueryData genProcesses(QueryContext &context) {
...
-      r["group"] = "0";
-      r["state"] = "0";
-      r["nice"] = "0";
-      r["uid"] = "-1";
-      r["gid"] = "-1";
-      r["euid"] = "-1";
-      r["egid"] = "-1";
-      r["suid"] = "-1";
-      r["sgid"] = "-1";
+      continue;
```
Comment (theopolis) [inline osquery/tables/system/darwin/processes.cpp]: Right, this will skip zombie processes. This is not good!

**Comment by osqueryer** [comment]
Build origin/pr/1888/merge (Job: 2423) is good! :+1:

File: osquery/tables/system/darwin/processes.cpp (hunk 1 above)
Comment (sharvilshah) [inline osquery/tables/system/darwin/processes.cpp]: For posterity: this is no longer the case and zombie processes will show up.

**Pull request merged**